File: services/openframe-ui/src/types/IntegratedTool.ts
export interface ToolCredentials {
  username?: string;
  password?: string;
  token?: string;
}

export interface IntegratedTool {
  id: string;
  name: string;
  description?: string;
  icon?: string;
  url?: string;
  port?: string;
  type?: string;
  toolType?: string;
  category?: string;
  platformCategory?: string;
  enabled: boolean;
  credentials?: ToolCredentials;
  
  // Layer information
  layer?: string;
  layerOrder?: number;
  layerColor?: string;
  
  // Monitoring configuration
  metricsPath?: string;
  healthCheckEndpoint?: string;
  healthCheckInterval?: number;
  connectionTimeout?: number;
  readTimeout?: number;
  allowedEndpoints?: string[];
} 

File: openframe/libs/openframe-core/src/main/java/com/openframe/core/model/IntegratedTool.java
package com.openframe.core.model;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.List;

@Data
@Builder
@Document(collection = "integrated_tools")
public class IntegratedTool {
    @Id
    private String id;
    private String name;
    private String description;
    private String icon;
    private List<ToolUrl> toolUrls;
    private String type;
    private String toolType;
    private String category;
    private String platformCategory;
    private boolean enabled;
    private ToolCredentials credentials;
    
    // Layer information
    private String layer;
    private Integer layerOrder;
    private String layerColor;
    
    // Monitoring configuration
    private String metricsPath;
    private String healthCheckEndpoint;
    private Integer healthCheckInterval;
    private Integer connectionTimeout;
    private Integer readTimeout;
    private String[] allowedEndpoints;
}